### PR TITLE
Added `make clean-docs` & consolidated Makefiles

### PR DESCRIPTION
**Pull request by ben-albrecht**
Added `make clean-docs` in main Makefile, which calls down to the sphinx Makefile. Cleaned up the main Makefile, and consolidated many rules into the 2 Makefiles inside the sphinx project.

### `/Makefile`
* `make clean-docs` cleans all dynamically generated files in `/doc/sphinx/`
* Many sphinx-related targets were migrated to `/doc/sphinx/Makefile`

### `/doc/sphinx/Makefile` & `/doc/sphinx/Makefile.sphinx`
* Split up into 2 makefile:
    * `Makefile.sphinx` contains sphinx-generated Makefile with minor modifications (comments/renames)
    * `Makefile` contains our own make rules, primarily `make symlink-docs` and a variety of `clean`s

### `/doc/sphinx/symlinks.py`
* Added argparse support (try `./symlinks.py --help` !)
* Added `--clean` flag to remove all symlinks

### `/doc/sphinx/make.bat`
* Removed because we require Windows users to use Cygwin  - this file was originally generated by sphinx

### CODE REVIEW

**Comment by ben-albrecht** [comment]
Summoning @Kyle-B @benharsh @ronawho

File: doc/sphinx/symlinks.py [hunk 1]
```diff
@@ -5,13 +5,35 @@
...
+    return opts
+
+
+
 
 def extfilter(directory, extension='.rst'):
     """Filter out files without extension '.rst'"""
     return [os.path.join(directory, f) for f in os.listdir(directory) if os.path.splitext(f)[1] == extension]
 
+def cleansymlinks():
```
Comment (Kyle-B) [inline doc/sphinx/symlinks.py]: Is this supposed to be here?

Comment (ben-albrecht) [inline doc/sphinx/symlinks.py]: Nope, just a remnant of me brainstorming good ways to implement an `undo symlinks`. Removed in last push.

File: Makefile [hunk 2]
```diff
@@ -127,13 +127,18 @@ module-docs: chpldoc
...
+doc-sphinx: module-docs
+	cd doc/sphinx && ${MAKE} symlink-docs
 	cd doc/sphinx && ./run-in-venv.bash ${MAKE} html
 
+clean-sphinx:
+	cd doc/shpinx && ${MAKE} cleanall
+
+docs: doc-sphinx
+
+clean-docs: FORCE
```
Comment (Kyle-B) [inline Makefile]: I don't think we need these clean rules (`clean-docs` and `clean-sphinx`) at the top level.

**Comment by Kyle-B** [comment]
Looks fine other than the comments I've made.

File: Makefile [hunk 3]
```diff
@@ -127,13 +127,18 @@ module-docs: chpldoc
...
-
-docs: module-docs symlink-docs
+doc-sphinx: module-docs
+	cd doc/sphinx && ${MAKE} symlink-docs
 	cd doc/sphinx && ./run-in-venv.bash ${MAKE} html
 
+clean-sphinx:
+	cd doc/shpinx && ${MAKE} cleanall
+
+docs: doc-sphinx
```
Comment (Kyle-B) [inline Makefile]: I'd merge this with `doc-sphinx`. and just leave it as `docs` to reduce the number of rules.

File: Makefile (hunk 2 above)
Comment (ben-albrecht) [inline Makefile]: Sure, if the user wants to clean only the docs, they will have to do it from inside `/doc`. Removed in latest pull - thanks for the review.

**Pull request merged**